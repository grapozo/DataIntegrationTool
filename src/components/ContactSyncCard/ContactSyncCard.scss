.contactSyncCard {
  background-color: #fff;
  border: 0.125rem solid #dfe3eb;
  border-radius: 0.125rem;
  padding: 4rem 2rem 4rem 2rem;
  text-align: center;

  .serviceName {
    color: #374a5e;
    font-size: 1.1rem;
    font-weight: 500;
  }

  .serviceDescription {
    color: #33475b;
    margin-top: 0.87rem;
    margin-bottom: 1.5rem;
    font-weight: 400;
  }
}
