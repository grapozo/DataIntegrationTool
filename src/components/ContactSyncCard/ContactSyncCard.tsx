import styles from './ContactSyncCard.scss';
import { ContactsDropdown } from '../ContactsDropdown';
import { Contact } from 'shared/interfaces/contact.interface';

type ContactsDropdownProps = {
  /** The icon of the service. This should be in svg format. */
  ServiceIcon: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
  /** The service name, such as Gmail or Mailchimp */
  serviceName: string;
  /** Label that describes what the sync should do. */
  serviceDescription: string;
  /** The contacts list. */
  contacts: Contact[];
};

/**
 * A card component which allows the user to select contacts to sync
 * with other services.
 */
export const ContactSyncCard = ({
  ServiceIcon,
  serviceName,
  serviceDescription,
  contacts
}: ContactsDropdownProps): JSX.Element => {
  return (
    <div className={`${styles.contactSyncCard}`}>
      <ServiceIcon />
      <div className={`${styles.serviceName}`}>{serviceName}</div>
      <div className={`${styles.serviceDescription}`}>{serviceDescription}</div>
      <ContactsDropdown contacts={contacts} />
    </div>
  );
};
