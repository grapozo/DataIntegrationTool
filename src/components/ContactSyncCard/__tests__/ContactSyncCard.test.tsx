import { create } from 'react-test-renderer';
import { ContactSyncCard } from '../ContactSyncCard';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

const IconMock = () => <svg>This is a svg mock</svg>;

const serviceName = 'Testing Card';
const serviceDescription = 'Testing the Sync Card!';

const selectedContactListChangedCallbackMock = jest.fn();

const MockContactSyncCard = () => (
  <ContactSyncCard
    ServiceIcon={IconMock}
    serviceName={serviceName}
    serviceDescription={serviceDescription}
    selectedContactListChangedCallback={selectedContactListChangedCallbackMock}
    contacts={[
      {
        id: 'ef1934cc85aec5ef909f74c7778529ea',
        name: 'Mocked Contact'
      },
      {
        id: 'ef1934cc85axc5ef909f3327178529ea',
        name: 'Mocked Contact 2'
      }
    ]}
  />
);

describe('ContactSyncCard', () => {
  it('should correctly render', () => {
    // This snapshot is mostly for reference, since any changes to
    // the tree will affect in this test.
    // In that case, we should just update the snapshot.
    const tree = create(<MockContactSyncCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should have the supplied svg icon on the screen', async () => {
    render(<MockContactSyncCard />);
    const svg = document.querySelector('svg');

    if (!svg) {
      fail('Should have the svg!');
    } else {
      expect(svg.innerHTML).toBe('This is a svg mock');
    }
  });
  it('should have the service name label', async () => {
    render(<MockContactSyncCard />);
    const serviceNameLabel = await screen.findByText(serviceName);
    expect(serviceNameLabel).toBeInTheDocument();
  });
  it('should have the service description label', async () => {
    render(<MockContactSyncCard />);
    const serviceDescriptionLabel = await screen.findByText(serviceDescription);
    expect(serviceDescriptionLabel).toBeInTheDocument();
  });
});
