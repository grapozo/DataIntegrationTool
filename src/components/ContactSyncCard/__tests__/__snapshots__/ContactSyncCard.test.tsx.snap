// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactSyncCard should correctly render 1`] = `
<div
  className="undefined"
>
  <svg>
    This is a svg mock
  </svg>
  <div
    className="undefined"
  >
    Testing Card
  </div>
  <div
    className="undefined"
  >
    Testing the Sync Card!
  </div>
  <div
    className="undefined "
  >
    <div
      className="
        undefined 
        
      "
      onClick={[Function]}
    >
      <span
        className="undefined"
      >
        No Contacts
      </span>
    </div>
    <div
      className="
      undefined 
      "
    >
      <div>
        <div
          className="undefined"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            Mocked Contact
          </label>
        </div>
      </div>
      <div>
        <div
          className="undefined"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            Mocked Contact 2
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
